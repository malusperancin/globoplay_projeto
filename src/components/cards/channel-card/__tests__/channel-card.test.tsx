/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import ChannelCardType from "../../channel-card/channel-card-type";
import ChannelCard from "../channel-card";

describe("ChannelCard", () => {
  const card: ChannelCardType = {
    backgroundUrl: "teste",
    description: "teste",
    headline: "teste",
    id: "teste",
    logo: "teste",
    logoSelected: "teste",
  };

  const channelComponent = (
    <ChannelCard data={card} isFocus={true} translate={100} />
  );

  const channelComponentNotFocus = (
    <ChannelCard data={card} isFocus={false} translate={100} />
  );

  it("should render without throwing an error", () => {
    const { container } = render(channelComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should be focus", () => {
    const { getByTestId } = render(channelComponent);
    const focus =
      getByTestId("channel-card").getElementsByClassName("buttonFocus")[0];

    expect(focus).toBeTruthy();
  });

  it("should not be focus", () => {
    const { getByTestId } = render(channelComponentNotFocus);
    const focus =
      getByTestId("channel-card").getElementsByClassName("button")[0];

    expect(focus).toBeTruthy();
  });
});
