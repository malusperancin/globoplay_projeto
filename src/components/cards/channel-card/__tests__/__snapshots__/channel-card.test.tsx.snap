// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="channel-card"
  >
    <button
      class="buttonFocus"
      style="transform: translate(100px);"
      type="button"
    >
      <img
        alt=""
        class="image"
        src="teste"
      />
    </button>
  </div>
</div>
`;
